app/src/main/java/org/p2p/wallet/newsend/model/CalculationMode.kt

```diff
@@ -149,15 +153,36 @@ class CalculationMode(private val sendModeProvider: SendModeProvider) {
         val tokenAround = usdAmount.divideSafe(token.usdRateOrZero, token.decimals)
         tokenAmount = tokenAround
 
-        handleCalculationUpdate(tokenAround.formatToken(), token.tokenSymbol)
+        handleCalculateTokenAmountUpdate()
     }
 
     private fun calculateByToken(inputAmount: String) {
         tokenAmount = inputAmount.toBigDecimalOrZero()
         usdAmount = tokenAmount.multiply(token.usdRateOrZero)
 
-        val usdAround = tokenAmount.times(token.usdRateOrZero)
-        handleCalculationUpdate(usdAround.formatUsd(), USD_READABLE_SYMBOL)
+        handleCalculateUsdAmountUpdate()
+    }
+
+    private fun handleCalculateTokenAmountUpdate() {
+        handleCalculationUpdate(
+            if (tokenAmount.lessThenMinValue()) {
+                lessThenMinString
+            } else {
+                tokenAmount.formatToken(token.decimals)
+            },
+            token.tokenSymbol
```

Comment:
You need to make Less than zero only for USD. 
The tokens has decimals up to 9 fraction